schemas/schema-kinds.md

```diff
@@ -1,26 +1,35 @@
 Schema Kinds
 ------------
 
-Recursive types contain additional type definitions for either their
-keys (in the case of maps),
-values (in the case of maps and lists),
+IPLD Schemas define a set of "kinds" that are built upon the
+[IPLD Data Model](https://github.com/ipld/specs/blob/master/data-model-layer/data-model.md#kinds):
+
+ - **Null**
+ - **Boolean**
+ - **Integer**
+ - **Float**
+ - **String**
+ - **Bytes**
+ - **List**
+ - **Map**
+ - **Link**
+ - **Union** - represented as a **map** in the data model for `keyed`,
+   `envelope` and `inline` representations, and varying data model kinds for
+   `kinded` unions, as described by [representations.md](representations.md).
+ - **Struct** - represented as a **map** in the data model by default but may be
+   used to describe **string** and **list** encodings, as described by
+   [representations.md](representations.md).
+ - **Enum** - represented as either a **string** or **int** in the data model,
+   as described by [representations.md](representations.md).
+ - **Copy** - a special kind that indicates that a type should be implemented
```

Comment:
Hmm.  Maybe `**copy**` should go in a separate list?

It shows up in the schema DSL (if not by name) and in the AST, but it's *not* going to show up in the [`schema.Kind` enum](https://github.com/ipld/go-ipld-prime/blob/53e4ebbe7bf632c9fe73baefee46135e923df8de/schema/kind.go#L13-L28) in the types that people actually use, which makes it a somewhat distinctive beast.